CHANGELOG.md

```diff
@@ -1,3 +1,31 @@
+## v1.20.0 (06/08/2024)
+
+**Common**
+* Updated links and filenames in the documentation.
+* Improved the CI scripts and pipelines.
+
+**olp-cpp-sdk-authentication**
+* Fixed the DateTime parser to respect DST timezones on iOS.
+* Fixed determining default credentials file absolute path on Windows for new distributions.
+
+**olp-cpp-sdk-core**
+* Extend logging with `olp::logging::LogContext` and `olp::logging::ScopedLogContext` to give users ability to manage logs better.
+* Added logging contexts usage to `olp::http::NetworkCurl` class.
+* Added a new error code `olp::client::ErrorCode::NoSpaceLeft` to signal shortage of available space on the target device.
+* To be able to propagate actual operation error, extended `olp::cache::KeyValueCache` and `olp::cache::DefaultCache` with methods `Read`, `Write`, `Delete` and `DeleteByPrefix`, which do the same actions as `Put`, `Get`, `Remove` and `RemoveKeysWithPrefix` correspondently, but return `olp::cache::OperationOutcome`.
+* Extended `olp::client::OlpClientSettings` with a new setting `propagate_all_cache_errors` to allow users enabling the propagation of all cache errors.
+* Added a new way to serialize a byte vector to JSON. 
+* Updated the logic to use generated session ID for the background network session for iOS.
+* Added support of Android-like build on desktop to be able to run tests on host PC.
+* Fixed deprecation warnings related to C++17.
+* Added background timeouts for iOS.
+* Made CMake option `OLP_SDK_ENABLE_IOS_BACKGROUND_DOWNLOAD` usable only for iOS.
+* Added missing thread names for heavy operations to ease profiling and debugging.
```

Comment:
"heavy operations" -> in what way heavy? Resource-heavy? CPU-intensive? Please clarify.

Reply:
I didn't specify a certain kind of operation because it implies all kinds of heavy operations, including CPU-intensive, RAM-heavy, disk-heavy, time-consuming etc.

Reply:
Done